hr_timesheet_role/models/account_analytic_line.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2018 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models, api, _
+from odoo.exceptions import ValidationError
+
+import logging
+_logger = logging.getLogger(__name__)
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = 'account.analytic.line'
+
+    role_id = fields.Many2one(
+        comodel_name='project.role',
+        string='Role',
+        required=False,
+        domain=lambda self: self._domain_role_id(),
+    )
+    is_role_required = fields.Boolean(
+        related='project_id.is_timesheet_role_required',
+    )
+    limit_role_to_assignments = fields.Boolean(
+        related='project_id.limit_timesheet_role_to_assignments',
```

Comment:
Same as above.

Reply:
This PR is based on #140 (and has commits of it), so please ignore the `hr_timesheet_role` part

Reply:
Yet you where correct that in hr_timesheet_sheet there are no `project_id` - so thanks a lot for a review!